frontend/test/lit/test_peephole_optimizations.py

```diff
@@ -75,6 +76,123 @@ def func():
 test_transform_named_sequence_injection()
 
 
+#
+# pipeline
+#
+
+
+def test_pipeline_lowering():
+    my_pipeline = {
+        "cancel_inverses": {},
+        "mitigate_with_zne": {
```

Comment:
Not sure about having ZNE in the pipeline like that. ZNE is not only applying a MLIR pass, there is stuff happening in the frontend as well. Otherwise why not to put the gradients as well?  In the pipeline should it not be optimization passes only? Also I think we restrict the ZNE to be applied on QNode. 

To my it is not clear what is the scope of the pipeline? We already have transform program in PL which is very similar. I think the pipeline should be a list of MLIR passes to apply on the function and not a list of transforms.

Reply:
Having ZNE is indeed quite painful implementation-wise, but it was specified in the epic (see PR description).

@josh146 does zne really need to be included in the pipeline? Personally I agree with Romain, that the pipeline should be restricted to "ZX-calculus"-style transformations.

Reply:
If we don't have ZNE, then it would be nice to have the merge rotation pass so that the pipeline can be meaningfully tested with more than one pass (right now there's just cancel inverses).

An alternative can be a empty peephole pass that does nothing (or inserts an identity, just for testing's sake).

Reply:
The merge rotation pass is planned for next quarter, so next iteration :+1: